sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -270,11 +260,141 @@ class AzureAppConfigurationProvider(Mapping[str, Union[str, JSON]]):
     keys. Enables resolution of Key Vault references in configuration settings.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, **kwargs) -> None:
         self._dict: Dict[str, str] = {}
         self._trim_prefixes: List[str] = []
         self._client: Optional[AzureAppConfigurationClient] = None
         self._secret_clients: Dict[str, SecretClient] = {}
+        self._key_vault_options: Optional[AzureAppConfigurationKeyVaultOptions] = kwargs.pop("key_vault_options", None)
+        self._selects: List[SettingSelector] = kwargs.pop(
+            "selects", [SettingSelector(key_filter="*", label_filter=EMPTY_LABEL)]
+        )
+        self._sentinel_keys: List[str] = []
+
+        trim_prefixes: List[str] = kwargs.pop("trim_prefixes", [])
+        self._trim_prefixes = sorted(trim_prefixes, key=len, reverse=True)
+        self._configuration_refresh = self._ConfigurationRefresh(**kwargs)
+
+    def refresh_configuration_settings(self, func):
+        @wraps(func)
+        def refresh_wrapper(*args, **kwargs):
+            self.refresh()
+            return func(*args, **kwargs)
+
+        return refresh_wrapper
+
+    def refresh(self, **kwargs) -> None:
```

Comment:
Did we decide to offer `Refresh` and `TryRefresh` separately?

Reply:
In the design meeting we decided to go with `refresh`

Reply:
Does `refresh` return true or false? Or does it just trigger the on_success/On_error callbacks?
Also, if nothing changed, will that trigger on_success or on_error callback?

Reply:
It doesn't return anything. `on_success` when a refresh happens. `on_error` if an error happens. Nothing for no refresh.